### PR TITLE
convert PrimitiveSimpleStore files to kotlin

### PR DESCRIPTION
**Pull request by davissuber**
This pull request converts `PrimitiveSimpleStore` and friends to kotlin. This is created with the assumption that we'd want to migrate away from the Future-based return type to Coroutines (and provide optional Future wrappings)

I am not sure how we are treating these conversions - I'm just using the auto java-to-kotlin conversion without worrying to much about making idiomatic kotlin changes to minimize mistakes.

### CODE REVIEW

File: simplestore/src/main/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreImpl.kt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    override fun deleteAllNow(): ListenableFuture<Void> {
+        return simpleStore.deleteAllNow()
+    }
+
+    override fun close() {
+        simpleStore.close()
+    }
+
+    override fun getInt(key: String?): ListenableFuture<Int?>? {
```
Comment (kurtisnelson) [inline simplestore/src/main/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreImpl.kt:62]: Key should not be nullable.

File: simplestore/src/main/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreFactory.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package com.uber.simplestore.primitive
+
+import com.uber.simplestore.DirectoryProvider
+import com.uber.simplestore.NamespaceConfig
+import com.uber.simplestore.impl.SimpleStoreFactory
+
+object PrimitiveSimpleStoreFactory {
+    @JvmOverloads
+    fun create(
+        directoryProvider: DirectoryProvider?, namespace: String?, config: NamespaceConfig? = NamespaceConfig.DEFAULT
```
Comment (kurtisnelson) [inline simplestore/src/main/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreFactory.kt:25]: No reason to make these nullable.

File: simplestore/src/test/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreTest.kt [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        SimpleStoreConfig.setIOExecutor(null)
+        SimpleStoreFactory.crashIfAnyOpen()
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun whenMissingOnDisk_numbers() {
+        create(directoryProvider, "", NamespaceConfig.DEFAULT).use { store ->
+            Truth.assertThat(store.contains(TEST_KEY).get()).isFalse()
+            val integer = store.getInt(TEST_KEY)!!.get()!!
```
Comment (kurtisnelson) [inline simplestore/src/test/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreTest.kt:46]: Can elvis these instead.

File: simplestore/src/test/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreTest.kt [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    fun reset() {
+        SimpleStoreConfig.setIOExecutor(null)
+        SimpleStoreFactory.crashIfAnyOpen()
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun whenMissingOnDisk_numbers() {
+        create(directoryProvider, "", NamespaceConfig.DEFAULT).use { store ->
+            Truth.assertThat(store.contains(TEST_KEY).get()).isFalse()
```
Comment (kurtisnelson) [inline simplestore/src/test/java/com/uber/simplestore/primitive/PrimitiveSimpleStoreTest.kt:45]: static import `assertThat`

**Comment by davissuber** [comment]
Sorry about this premature pull request - after re-examining the class structures, I now think it's probably better to convert `SimpleStore` first to avoid some of the "kotlin assumes a non-null thing to be nullable because the corresponding java type is ambiguous" situation.

Also, taking a step back, if the end goal is to embrace coroutine usage - it's worth zooming out to think what is the overall migration strategy and we may or may not want to leave the current Future stuff untouched at first (?)

Will re-introduce/reopen after more deliberation.

**Pull request closed without merging**